### ISSUE TITLE
Problem loading a camera

### ISSUE THREAD

**Issue by LMSC-NTappy** [PyMoDAQ/pymodaq_plugins_thorlabs#4]
Hello, 

I am facing an error initialize a thorlabs camera. The Camera is correctly detected 
``` 
...
[datetime] pymodaq.pymodaq.daq_viewer_main.Testing - INFO - Ready
[datetime] pymodaq.pymodaq.daq_viewer_main.Testing - INFO -   File "C:\Users\admin\.conda\envs\sp_dev\lib\site-packages  
\pymodaq_plugins_thorlabs\daq_viewer_plugins\plugins_2D\daq_2Dviewer_Thorlabs_DCx.py", line 91, in ini_detector
    self.settings.child('exposure').setValue(self.controller.exposure.m_as('ms'))
'UC480_Camera' object has no attribute 'exposure'
[datetime]pymodaq.pymodaq.daq_viewer_main.Testing - INFO - detector initialized: False
```

I think the fix might be as easy as replacing 
https://github.com/CEMES-CNRS/pymodaq_plugins_thorlabs/blob/5df60c9fea7b6f41546dd1b3ea8d3732d351e3e0/src/pymodaq_plugins_thorlabs/daq_viewer_plugins/plugins_2D/daq_2Dviewer_Thorlabs_DCx.py#L91

and other similar instances such as

https://github.com/CEMES-CNRS/pymodaq_plugins_thorlabs/blob/5df60c9fea7b6f41546dd1b3ea8d3732d351e3e0/src/pymodaq_plugins_thorlabs/daq_viewer_plugins/plugins_2D/daq_2Dviewer_Thorlabs_DCx.py#L54

by `self.controller._get_exposure()` and `self.controller._set_exposure(Q_(...))`  

I am unable to submit a PR right now since I don't manage to get this plugin installed from source, see  
CEMES-CNRS/PyMoDAQ#71 but I hope to get it right at some point lol

**Comment by seb5g** [comment]
Hello,
for this plugin I'm using the package Instrumental that at the time was missing some functionalities. I did a PR from my fork but it seems it is not yet included in the official branch. You can therefore try to install Instrumental from :

https://github.com/CEMES-CNRS/Instrumental

see pip documentation on how to install from a git url! 

However Instrumental also requires the thorlabs library to be processed once for the python code to work. If you cannot find how to do it, you can try to execute the module __build_uc480.py_:
![image](https://user-images.githubusercontent.com/6609955/151762950-ee9f5ea3-278d-415c-ac11-173b5f8341d9.png)


Let me know if it works

**Comment by LMSC-NTappy** [comment]
Hello,

Did you have success controlling the camera exposure time with your implementation? 

I have checked mabuchilab/Instrumental#144 and in my opinion there is something broken beyond the missing `exposure ` facet. 

See mabuchilab/Instrumental#150

**Comment by seb5g** [comment]
Well I got that camera only for a short time to produce the new PID module and the video on it. I don't remember if I made it finally working but I think so. @quantumm is the owner of the camera, so maybe he can do some testing?

**Comment by LMSC-NTappy** [comment]
If he can, that'd be great! 

In any case, I've opened #5 on the topic. For me it currently works so if @quantumm finds some time to test it'd be great

**Comment by quantumm** [comment]
Hello,
I will try to do those tests but unfortunately for now it is difficult to find some time :/

**Comment by quantumm** [comment]
Hi,

I tried again to install the plugin to use a DCC camera with a fresh install of pymodaq (3.5.6).
I was surprised to see that this issue is still not solved.
It seems like something wrong happened during the merge of the PR #5 .
Such call like `controller.exposure` should have been removed, which is not the case if you look at lign 91 in the current version of this file : [https://github.com/CEMES-CNRS/pymodaq_plugins_thorlabs/blob/0.1.2/src/pymodaq_plugins_thorlabs/daq_viewer_plugins/plugins_2D/daq_2Dviewer_Thorlabs_DCx.py](https://github.com/CEMES-CNRS/pymodaq_plugins_thorlabs/blob/0.1.2/src/pymodaq_plugins_thorlabs/daq_viewer_plugins/plugins_2D/daq_2Dviewer_Thorlabs_DCx.py)

**Comment by LMSC-NTappy** [comment]
Hi 

The merge is fine, the changes have just not been released in a new version of the plugin yet. 

For now, you should still install from source the main branch of this PR to obtain theses fixes.

**Comment by quantumm** [comment]
Ah ok. Think I got it ;) Thanks.

**Comment by quantumm** [comment]
I only tested the camera, not the other instruments, but it is fine for that part yes.

**Comment by LMSC-NTappy** [comment]
I think this can be achieved directly from the install_requires if you have a clone of instrumental_lib on github that will reliably work for an extended period of time:

https://stackoverflow.com/questions/32688688/how-to-write-setup-py-to-include-a-git-repository-as-a-dependency

Cheers

Nicolas

**Comment by seb5g** [comment]
FOUND IT !!!

You have to put this in the requirements (in your plugin toml file): 

[plugin-install]
#packages required for your plugin:
packages-required = ["pythonnet", "pywin32", "nicelib", "instrumental-lib @ https://github.com/CEMES-CNRS/Instrumental/archive/refs/tags/cemes.0.6.tar.gz"]

so you can point directly to an archive file without need of git. You could also do it for your dependencies on pylablib also

**Comment by seb5g** [comment]
I will therefore publish a new version of thorlabs plugin including this requirement!

**Comment by rgeneaux** [comment]
Hi guys,


Just wanted to note that the new release of the plugin didn't get uploaded to pypi
 
( error in upload workflow)
```
" Invalid value for requires_dist. Error: Can't have direct dependency:  
         'instrumental-lib @ https://github.com/CEMES-CNRS/Instrumental/archive/
         refs/tags/cemes.0.6.tar.gz' 
```

so at the moment, it cannot be installed by the plugin manager. 
I was running into the camera exposure issue and it kind of drove me crazy until I noticed that I wasn't on the last release :-)

**Comment by seb5g** [comment]
Ok I reverted the requirements on instrumental-lib for the moment and published v 0.2.0 on pypi manually. I'm in the process of a push release on instrumental so should have the next release soon to make it all work.
